### PR TITLE
Parse file subpath from the relative path

### PR DESCRIPTION
**Pull request by matszczygiel**
This is a fix for Windows issues when sending the directories. The `FileSubPath` was parsed incorrectly from the filesystem path

### CODE REVIEW

File: norddrop/src/device/mod.rs [hunk 1]
```diff
@@ -406,8 +406,8 @@ fn prepare_transfer_files(
 ) -> Result<Vec<File>> {
     let mut files = Vec::new();
 
+    #[allow(unused_variables)]
```
Comment (LukasPukenis) [inline norddrop/src/device/mod.rs:409]: why was this moved?

File: drop-transfer/src/file/id.rs [hunk 2]
```diff
@@ -92,6 +92,15 @@ impl FileSubPath {
...
     }
+
+    pub fn from_path(path: impl AsRef<Path>) -> crate::Result<Self> {
+        let vec = path
+            .as_ref()
+            .iter()
+            .map(|cmp| cmp.to_str().map(String::from).ok_or(crate::Error::BadPath))
+            .collect::<Result<_, _>>()?;
+        Ok(Self(vec))
+    }
```
Comment (LukasPukenis) [inline drop-transfer/src/file/id.rs:103]: Could you provide more context on what this does and what was wrong with the previous approach?

File: norddrop/src/device/mod.rs (hunk 1 above)
Comment (matszczygiel) [reply norddrop/src/device/mod.rs:409]: Because of `i` in the for loop, it's unused on Windows and generated a warning without this change

File: drop-transfer/src/file/id.rs (hunk 2 above)
Comment (matszczygiel) [reply drop-transfer/src/file/id.rs:103]: This one constructs the `FileSubPath` from path components. Previously we only had a method that constructed it by splitting the path string on `/` which is clearly incorrect for Windows.
I mean both methods are needed because the string representation used in the WS protocol is always using `/` but when constructing from the actual filesystem path we need to use this method

Comment (LukasPukenis) [reply drop-transfer/src/file/id.rs:103]: Thanks for explaining, make sense

**Review by LukasPukenis — APPROVED** [review]
